### PR TITLE
Refactor/Migrate to new Relationship APIs

### PR DESCRIPTION
**Pull request by Magnus-Kuhn**

### CODE REVIEW

File: packages/enmeshed_types/lib/src/contents/relationship_creation_content/relationship_creation_content.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  Map<String, dynamic> toJson();
+
+  @mustCallSuper
+  @override
+  List<Object?> get props;
+}
+
+class ArbitraryRelationshipCreationContent extends RelationshipCreationContent with MapMixin<String, dynamic>, ArbitraryJSON {
```
Comment (britsta) [inline packages/enmeshed_types/lib/src/contents/relationship_creation_content/relationship_creation_content.dart:29]: This isn't the `ArbitraryRelationshipCreationContent`, which will only be introduced by the (currently not yet merged) PR https://github.com/nmshd/runtime/pull/170, right?

File: packages/enmeshed_types/lib/src/dtos/relationship.dart [hunk 2]
```diff
@@ -1,46 +1,102 @@
...
+        'creationContent': creationContent.toJson(),
+        'auditLog': auditLog.map((e) => e.toJson()).toList(),
       };
 
   @override
-  List<Object?> get props => [id, template, status, peer, peerIdentity, changes];
+  List<Object?> get props => [id, template, status, peer, peerIdentity, creationContent, auditLog];
+}
+
+class RelationshipAuditLogEntryDTO extends Equatable {
```
Comment (britsta) [inline packages/enmeshed_types/lib/src/dtos/relationship.dart:63]: `RelationshipAuditLogDTO` is missing.

File: .dev/compose.backbone.env [hunk 3]
```diff
@@ -1 +1 @@
-BACKBONE_VERSION=5.9.1
+BACKBONE_VERSION=6.2.0
```
[the remark this answers is not in the document]
Comment (britsta) [reply .dev/compose.backbone.env:1]: Okay, https://github.com/nmshd/runtime/pull/125/files#diff-0cff58168f80528b3a79e26b82851cdd8b3b24b6d8ba5ce750a01b715012e4f1 is merged now, so this comment is a little bit outdated :D As this PR did not take the changes from `5.0.0-alpha.4` to `5.0.0-alpha.6` (removal of `mustBeAccepted` property of RequestItemGroup and decomposition of Relationships) into account, I still think the `BACKBONE_VERSION` should be changed to `6.1.0` here and the Runtime version in the `package.json` from `5.0.0-alpha.5` to `5.0.0-alpha.4`.

File: packages/enmeshed_runtime_bridge/integration_test/services/facades/transport/account_facade_test.dart [hunk 4]
```diff
@@ -23,9 +23,8 @@ void run(EnmeshedRuntime runtime) {
       final identityInfoResult = await session.transportServices.account.getIdentityInfo();
 
       expect(identityInfoResult, isSuccessful<GetIdentityInfoResponse>());
-      expect(identityInfoResult.value.address.length, lessThanOrEqualTo(36));
```
[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/enmeshed_runtime_bridge/integration_test/services/facades/transport/account_facade_test.dart:26]: I think we should expect addresses completely different. For example splitting it at : and only looking at the last (address) part.

File: packages/enmeshed_runtime_bridge/integration_test/services/facades/transport/messages_facade_test.dart [hunk 5]
```diff
@@ -192,7 +192,7 @@ void run(EnmeshedRuntime runtime) {
   });
 
   group('Message errors', () {
-    const fakeAddress = 'id1PNvUP4jHD74qo6usnWNoaFGFf33MXZi6c';
+    const fakeAddress = 'did:e:prod.enmeshed.eu:dids:fef1992c5e529adc413288';
```
[the remark this answers is not in the document]
Comment (Magnus-Kuhn) [reply packages/enmeshed_runtime_bridge/integration_test/services/facades/transport/messages_facade_test.dart:195]: I guess the cat is out of the bag there (why did I do this when it wasn't necessary ???)

[the remark this answers is not in the document]
Comment (Magnus-Kuhn) [reply packages/enmeshed_runtime_bridge/integration_test/services/facades/transport/messages_facade_test.dart:195]: I'll include the runtime change in my next PR

File: packages/enmeshed_types/lib/src/contents/relationship_creation_content/relationship_creation_content.dart (hunk 1 above)
Comment (Magnus-Kuhn) [reply packages/enmeshed_types/lib/src/contents/relationship_creation_content/relationship_creation_content.dart:29]: right, arbitraryContents are also used elsewhere

File: packages/enmeshed_types/test/contents/token_content/token_content_device_shared_secret_test.dart
[the remark this answers is not in the document]
Comment (Magnus-Kuhn) [reply packages/enmeshed_types/test/contents/token_content/token_content_device_shared_secret_test.dart:1]: removed it

File: packages/enmeshed_runtime_bridge/integration_test/services/facades/consumption/attributes_facade_test.dart
[the remark this answers is not in the document]
Comment (Magnus-Kuhn) [reply packages/enmeshed_runtime_bridge/integration_test/services/facades/consumption/attributes_facade_test.dart:1]: I'd like to remark that I hate facades exposing use cases that are only supposed to be used through the request module for additional implementation effort.

[the remark this answers is not in the document]
Comment (britsta) [reply packages/enmeshed_runtime_bridge/integration_test/services/facades/consumption/attributes_facade_test.dart:1]: As I understand it, the use cases should also be able to be executed by Integrators of Connectors if they do not want to use the Request module and want to implement the processes themselves instead, even though the Request module will probably be active most of the time in practice. However, I do not know exactly whether it makes sense to transfer such use cases from a facade of the Runtime to a facade of the App.

[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/enmeshed_runtime_bridge/integration_test/services/facades/consumption/attributes_facade_test.dart:1]: Please do not add methods we don't need! We should not use `deleteSharedAttributesForRejectedOrRevokedRelationship` and don't need to expose this here!

File: packages/enmeshed_runtime_bridge/integration_test/services/facades/transport/account_facade_test.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (Magnus-Kuhn) [reply packages/enmeshed_runtime_bridge/integration_test/services/facades/transport/account_facade_test.dart:26]: I'm going full regex match

File: packages/enmeshed_types/lib/src/dtos/relationship_change.dart
[the remark this answers is not in the document]
Comment (Magnus-Kuhn) [reply packages/enmeshed_types/lib/src/dtos/relationship_change.dart:1]: I'm not sure about the end result with the texts - would the app devs like us to leave them some TODOs where they need to add something?

File: packages/enmeshed_types/lib/src/dtos/relationship.dart (hunk 2 above)
Comment (Magnus-Kuhn) [reply packages/enmeshed_types/lib/src/dtos/relationship.dart:63]: This setup with classes doesn't allow an unnamed property List<AuditLogEntry>. If consistency is needed, I'm happy to ditch it in the runtime.

Comment (jkoenig134) [reply packages/enmeshed_types/lib/src/dtos/relationship.dart:63]: `RelationshipAuditLogDTO` ist just a list of `RelationshipAuditLogEntryDTO` and don't has to have an equivalent in this coding

File: packages/enmeshed_types/lib/src/contents/attribute_query/third_party_relationship_attribute_query.dart [hunk 6]
```diff
@@ -1,9 +1,10 @@
 part of 'attribute_query.dart';
 
-@JsonSerializable(includeIfNull: false)
+enum ThirdPartyRelationshipAttributeQueryOwner { thirdParty, recipient, empty }
+
 class ThirdPartyRelationshipAttributeQuery extends AttributeQuery {
```
Comment (jkoenig134) [inline packages/enmeshed_types/lib/src/contents/attribute_query/third_party_relationship_attribute_query.dart:5]: plase make this JsonSerializable again

File: packages/enmeshed_types/lib/src/contents/relationship_creation_content/relationship_creation_content_containing_response.dart [hunk 7]
```diff
@@ -0,0 +1,24 @@
+part of 'relationship_creation_content.dart';
+
+class RelationshipCreationContentContainingResponse extends RelationshipCreationContent {
```
Comment (jkoenig134) [inline packages/enmeshed_types/lib/src/contents/relationship_creation_content/relationship_creation_content_containing_response.dart:3]: RelationshipCreationContent

File: packages/enmeshed_types/lib/src/contents/relationship_creation_content/relationship_creation_content.dart [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+import 'package:equatable/equatable.dart';
+import 'package:meta/meta.dart';
+
+import '../arbitraty_json.dart';
+import '../response.dart';
+
+part 'relationship_creation_content_containing_response.dart';
+
+abstract class RelationshipCreationContent extends Equatable {
```
Comment (jkoenig134) [inline packages/enmeshed_types/lib/src/contents/relationship_creation_content/relationship_creation_content.dart:11]:
```suggestion
abstract class RelationshipCreationContentDerivation extends Equatable {
```

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: packages/enmeshed_types/lib/src/dtos/relationship_change.dart
[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/enmeshed_types/lib/src/dtos/relationship_change.dart:1]: @britsta please keep that in mind for the final review of the v5 branch.

**Review by jkoenig134 — APPROVED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**